pkg/transforms/policy.go

```diff
@@ -130,16 +130,172 @@ func OperatorPolicyResourceBuilder(c *unstructured.Unstructured) *PolicyResource
+
+	return &PolicyResource{
+		node: node,
+	}
+}
+
+func recordRelatedObjects(c *unstructured.Unstructured, node Node) Node {
+	relatedObjects, found, err := unstructured.NestedSlice(c.Object, "status", "relatedObjects")
+	if !found || err != nil {
+		return node
+	}
+
+	objList := make([]policyRelObj, 0, len(relatedObjects))
+
+	for _, item := range relatedObjects {
+		relObj := parseRelatedObject(item)
+		if relObj == nil {
+			continue
+		}
+
+		objList = append(objList, *relObj)
+	}
+
+	slices.SortFunc(objList, func(a, b policyRelObj) int {
+		return strings.Compare(a.kind+a.namespace+a.name, b.kind+b.namespace+b.name)
+	})
+
+	node.Metadata["relObjs"] = objList
+
+	return node
+}
+
+func parseRelatedObject(item any) *policyRelObj {
+	relObj, ok := item.(map[string]any)
+	if !ok {
+		return nil
+	}
+
+	object, found, err := unstructured.NestedMap(relObj, "object")
+	if !found || err != nil {
+		return nil
+	}
+
+	meta, found, err := unstructured.NestedStringMap(object, "metadata")
+	if !found || err != nil {
+		return nil
+	}
+
+	kind, ok := object["kind"].(string)
+	if !ok {
+		return nil
+	}
+
+	return &policyRelObj{
+		kind:      kind,
+		namespace: meta["namespace"],
+		name:      meta["name"],
+	}
+}
+
+func CertPolicyResourceBuilder(c *unstructured.Unstructured) *PolicyResource {
+	node := transformCommon(c) // Start off with the common properties
+
+	detailMap, found, err := unstructured.NestedMap(c.Object, "status", "compliancyDetails")
+	if len(detailMap) == 0 || !found || err != nil {
+		return &PolicyResource{
+			node: getPolicyCommonProperties(c, node),
+		}
+	}
+
+	certList := []policyRelObj{}
+	for namespace, item := range detailMap {
+		details, ok := item.(map[string]any)
+		if !ok {
+			continue
+		}
+
+		// this "list" is actually a map
+		nonCompCerts, found, err := unstructured.NestedMap(details, "nonCompliantCertificatesList")
+		if len(nonCompCerts) == 0 || !found || err != nil {
+			continue
+		}
+
+		for _, item := range nonCompCerts {
+			cert, ok := item.(map[string]any)
+			if !ok {
+				continue
+			}
+
+			name, ok := cert["secretName"].(string)
+			if !ok {
+				continue
+			}
+
+			certList = append(certList, policyRelObj{
+				kind:      "Secret",
+				namespace: namespace,
+				name:      name,
+			})
+		}
+	}
+
+	slices.SortFunc(certList, func(a, b policyRelObj) int {
+		return strings.Compare(a.kind+a.namespace+a.name, b.kind+b.namespace+b.name)
+	})
+
+	node.Metadata["relObjs"] = certList
+
+	return &PolicyResource{
+		node: getPolicyCommonProperties(c, node),
+	}
 }
 
 // BuildNode construct the node for Policy Resources
 func (p PolicyResource) BuildNode() Node {
 	return p.node
 }
 
-// BuildEdges construct the edges for Policy Resources
 func (p PolicyResource) BuildEdges(ns NodeStore) []Edge {
-	// no op for now to implement interface
-	return []Edge{}
+	kindRaw, ok := p.node.Properties["kind"]
+	if !ok {
+		return []Edge{}
+	}
+
+	policyKind, ok := kindRaw.(string)
+	if !ok {
+		return []Edge{}
+	}
```

Comment:
You can combine these 2 and save a few lines.

```suggestion
	policyKind, ok := p.node.Properties["kind"].(string)
	if !ok {
		return []Edge{}
	}
```